doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_recovery.rst

```diff
@@ -0,0 +1,183 @@
+Still, some bugs can be missed during the development phase, which could lead to inconsitencies in the firmware.
+
+How the recovery mode and recovery manifests work
+*************************************************
+
+The recovery manifests form a separate hierarchy from the normal manifests.
+In this hierarchy the application recovery (APP_RECOVERY) manifest has both the responsibility of managing the application core image as well as managing other manifests such as the radio recovery manifest.
+
+If a failure during a boot process occured, the Secure Domain sets the recovery flag and reboots the device.
+Upon each boot the Secure Domain checks if the recovery flag is set.
+If it is set, the device enters recovery mode.
+If it isn't booting proceeds normally by running the root manifest.
+
+After entering the recovery mode it is verified if MPI configuration for the APP_RECOVERY is present.
+If it is, the APP_RECOVERY manifest is processed.
+It no MPI configuration is found the Secure Domain performs an attempt to process the normal manufacturer root manifest.
+This is needed, as the device might enter recovery mode if it is empty.
+The recovery flag might not be cleared after flashing the firmware, but the device should proceed as if it would boot normally.
+
+The role of the recovery application is to perform an update of the main application firmware, which does not differ from the normal SUIT update process.
+As soon as the update finishes succesfully, the recovery flag is cleared and the device proceeds with normal operation.
+
+Default recovery firmware
+*************************
+
+Nordic provides a default recovery firmware that can be used in the recovery process.
+This firmware uses Bluetooth LE and SMP as a transport.
+It is optimized for memory usage, currently using around 164 kB of MRAM (72 kB of application core and 92 kB radio core).
+
+.. note::
+   The default recovery firmware does not support updating from external flash memory.
+
+To use the firmware:
+
+1. Create :file:`recovery.overlay` and :file:`recovery_hci_ipc.ovelay` files in the main application's :ref:`configuration_system_overview_sysbuild` directory.
+   These files must define the ``cpuapp_recovery_partition`` and ``cpurad_recovery_partition`` nodes respectively.
+   These partitions specify where the images for the recovery firmware are stored and cannot overlap with the main application partitions.
+   For reference, see the files in the ``samples/suit/smp_transfer`` sample.
+
+#. Set the ``SB_CONFIG_SUIT_BUILD_RECOVERY`` sysbuild configuration option in the main application.
+   This will cause the recovery firmware to be built automatically as part of the main application build.
+
+#. Flash the main application firmware to the device:
+
+      .. code-block:: bash
+
+         west flash
+
+   This will automatically flash both the main application and the recovery firmware to the device.
```

Comment:
```suggestion
#. :ref:`Program the main application firmware to the device <programming>`.
   This will automatically program both the main application and the recovery firmware to the device.
```
Not `west flash --recover`?